### PR TITLE
Implement Struct Field, and allow Field to use it

### PR DESCRIPTION
**Pull request by maxice8**
- Implement StructField
- Add support for StructField inside Field

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: vespa/package.py [hunk 1]
```diff
@@ -247,7 +338,8 @@ def __repr__(self):
             repr(self.match),
             repr(self.weight),
             repr(self.bolding),
-            repr(self.summary)
+            repr(self.summary),
+            repr(self.struct_fields),
```
Comment (lesters) [inline vespa/package.py:342]: I guess this PR conflicts with https://github.com/vespa-engine/pyvespa/pull/411 and https://github.com/vespa-engine/pyvespa/pull/412 and https://github.com/vespa-engine/pyvespa/pull/410 here?

**Review by lesters — APPROVED** [review]
LGTM, but I think some work needs to be done before merging due to the other PRs wirking on fields.

File: vespa/package.py (hunk 1 above)
Comment (maxice8) [reply vespa/package.py:342]: Yes, each of my PRs (that the Field class will conflict upon merging either of them), in any case I think we should deal with the [amount of constructors issue](https://github.com/vespa-engine/pyvespa/pull/412#discussion_r1066205019) before merging any of them.

**Comment by maxice8** [comment]
Rebased on top of master

**Pull request merged**